### PR TITLE
make additional tests compatible with Jenkins 2.307

### PR DESCRIPTION
**Pull request by car-roll**
additional test cleanup to support the terminology changes in 2.307
downstream of #168 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java [hunk 1]
```diff
@@ -809,7 +809,7 @@ private static void finish(@CheckForNull final String cookie) {
                         env.put(COOKIE_VAR, cookie);
                         // Cf. CoreEnvironmentContributor:
                         if (exec.getOwner() instanceof MasterComputer) {
-                            env.put("NODE_NAME", "master");
+                            env.put("NODE_NAME", node.getSelfLabel().getName()); // mirror https://github.com/jenkinsci/jenkins/blob/89d334145d2755f74f82aad07b5df4119d7fa6ce/core/src/main/java/jenkins/model/CoreEnvironmentContributor.java#L63
```
Comment (car-roll) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java:812]: Copied from #168

**Review by timja — APPROVED** [review]
🚢

**Review by dwnusbaum — APPROVED** [review]
Looks fine to me. I would test it with `mvn clean verify -Djenkins.version=2.307` if you haven't already just to make sure there aren't any issues.

**Comment by timja** [comment]
> Looks fine to me. I would test it with `mvn clean verify -Djenkins.version=2.307` if you haven't already just to make sure there aren't any issues.

I ran it locally on:

(Mac M1 ARM64)
```
openjdk version "11.0.12" 2021-07-20 LTS
```

On both current core and new core result was:
```
 Tests run: 79, Failures: 3, Errors: 0, Skipped: 4
```

```
[ERROR] Failures:
[ERROR]   ShellStepTest.abort:213->ensureForWhile:781 org.jenkinsci.plugins.workflow.steps.durable_task.ShellStepTest$$Lambda$448/0x0000000800d91040@7e656bc7
[ERROR]   ShellStepTest.interruptingAbortsBuild:723 unexpected build status
[ERROR] org.jenkinsci.plugins.workflow.steps.durable_task.ShellStepTest.interruptingAbortsBuildEvenWithReturnStatus
```

Tests failed for:
```
Expected: is <ABORTED>
     but: was <SUCCESS>
```

**Comment by car-roll** [comment]
@timja @dwnusbaum I get the same failures locally, but on CI those tests always pass. The failing tests for 2.307+ are: `ShellStepTest.missingNewline`, `ShellStepTest.remoteConsoleNotes`, `ShellStepTest.remoteLogger`.
I tested those under 2.307 and confirmed they failed before this PR and succeeded after.

**Comment by timja** [comment]
> Though this issue locally feels like it's related to [issues.jenkins.io/browse/JENKINS-65195](https://issues.jenkins.io/browse/JENKINS-65195) ?

Could be yeah there's other process issues that aren't fixed, e.g. if I go to /restart in the browser and click restart I'll get a 'killed' message in my terminal and it doesn't restart just dies.

**Pull request merged**